### PR TITLE
Feature/exercise fragments

### PR DESCRIPTION
**Pull request by PKlumpp**
Merging the most recent version of exercise Fragments.

The package now contains a separate fragment ExerciseMenu which allows the user to chose between going to the next exercise, or redo the previous. Additionally, the architecture was implemented more generic, the hosting activity only has to implement two interfaces now. One for the Exercise Fragments response, and a second one for the ExerciseMenu response.

### CODE REVIEW

File: app/src/main/java/com/sma2/sma2/ExerciseFragments/ExerciseFragment.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        }
+    }
+
+    @Override
+    public void onDetach() {
+        super.onDetach();
+        mListener = null;
+    }
+
+    public interface OnFragmentInteractionListener {
```
Comment (AKuederle) [inline app/src/main/java/com/sma2/sma2/ExerciseFragments/ExerciseFragment.java]: Maybe we can have a better name for this.

Comment (PKlumpp) [reply app/src/main/java/com/sma2/sma2/ExerciseFragments/ExerciseFragment.java]: Das stimmt. Wie wäre es mit OnExerciseCompletedListener? Der wird von mir ja nur gefeuert wenn die Exercise beendet ist.

Comment (AKuederle) [reply app/src/main/java/com/sma2/sma2/ExerciseFragments/ExerciseFragment.java]: Jap Klingt gut

**Comment by AKuederle** [comment]
@PKlumpp kannst du die Konflikts beseitigen? Dann merge ich das

**Pull request merged**